lib/stringify.js

```diff
@@ -102,6 +102,14 @@ var stringify = function stringify(
     if (isNonNullishPrimitive(obj) || utils.isBuffer(obj)) {
         if (encoder) {
             var keyValue = encodeValuesOnly ? prefix : encoder(prefix, defaults.encoder, charset, 'key', format);
+            if(generateArrayPrefix === 'comma' && encodeValuesOnly){
+                const valuesArray = String(obj).split(',');
```

Comment:
`String(obj)` for a non-array object will turn into `[object Object]`, which i don't think is what we want.